tests/tests/test_models.py

```diff
@@ -103,3 +107,97 @@ def test_delete_old(self):
         messages = Message.objects.all()
         self.assertEqual(len(messages), 1)
         self.assertGreaterEqual(messages[0].date_created, timezone.now() - timedelta(days=days))
+
+    @patch("django.core.mail.message.generator.Generator._make_boundary")
+    def test_get_email_message_roundtrip(self, mock_make_boundary):
+        self.maxDiff = None
+
+        def get_boundary(token: int):
+            return ('=' * 15) + (_fmt % token) + '=='
+
+        boundaries = [get_boundary(10), get_boundary(20), get_boundary(30), get_boundary(40)]
+
+        def reset_mock():
+            assert mock_make_boundary.call_count == 2
+            mock_make_boundary.reset_mock(side_effect=True)
+            mock_make_boundary.side_effect = boundaries
+
+        mock_make_boundary.side_effect = boundaries
+
+        ref = EmailMultiAlternatives(
+            subject="Some subject",
+            body="Plain text body",
+            from_email="from@example.com",
+            to=["to@example.com"],
+            bcc=["bcc@example.com"],
+            cc=["cc@example.com"],
+            headers={"X-Custom": "custom-header"},
+            reply_to=["reply-to@example.com"],
+        )
+        ref.attach_alternative(
+            "<html><body><img src=\"cid:123456789\" /></body></html>",
+            "text/html"
+        )
+        ref.mixed_subtype = "related"
+        # add an inline cid image (PNG, white pixel)
+        image = MIMEImage(
+            b"\x89PNG\r\n\x1a\n\x00\x00\x00\rIHDR\x00\x00\x00\x01\x00\x00\x00\x01\x08"
+            b"\x06\x00\x00\x00\x1f\x15\xc4\x89\x00\x00\x00\x01sRGB\x00\xae\xce\x1c\xe9"
+            b"\x00\x00\x00\rIDAT\x18Wc\xf8\xff\xef\xdf\x7f\x00\t\xf6\x03\xfbW\xfe{\x1b"
+            b"\x00\x00\x00\x00IEND\xaeB`\x82"
+        )
+        image.add_header("Content-ID", f"<123456789>")
```

Comment:
This is a bit nit, but flake8 is complaining about the f-string: `f-string is missing placeholdersflake8(F541)`.

Reply:
Yes, we have considered adding at least flake8 and we will probably do it in the future. Not 100% sure about adding black and reformat all the code.